File: product_front_end/src/Component/Product.js
import React from "react";
import { useState, useEffect } from "react";
import { useParams } from "react-router-dom";
export function Product() {
  const [product, setProduct] = useState({});
  const { id } = useParams();
  useEffect(() => {
    fetch("https://localhost:7107/api/Product/" + id)
      .then((res) => res.json())
      .then((result) => {
        setProduct(result);
      });
  }, {});
  return (
    <div>
      <label>Id:</label>
      {product.id}
      <br />
      <label>Category:</label>
      {product.category}
      <br />
      <label>Name:</label>
      {product.name}
      <br />
      <label>Price:</label>
      {product.price}
      <br />
      <label>MinQty:</label>
      {product.minQty}
    </div>
  );
}
export default Product;

File: Product/Models/Product.cs
namespace Product.Models
{
    public class Product
    {
        public int Id { get; set; }
        public String? Category {  get; set; }
        public String? Name { get; set; }
        public Double? Price { get; set; } 
        public int? MinQty { get; set; }
    }
}
